src/qml/SoundPage.qml

```diff
@@ -99,15 +83,7 @@ Item {
             height: Dims.h(25)
             stepSize: 10
             value: volumeControl.volume
-            onValueChanged: {
-                volumeControl.volume = value
-                // Un-mute if muted
-                if (preMuteLevel.value > 0) {
-                    // Restore pre mute volume value
-                    volumeControl.volume = preMuteLevel.value
-                    preMuteLevel.value = 0
-                }
-            }
+            onValueChanged: [ volumeControl.volume, preMuteLevel.value ] = [ value, 0 ]
```

Comment:
This doesn't actually restore the previous volume, instead it just starts from `0` again. I'm not sure if this was the intended behavior. @eLtMosen can you comment on this?

Reply:
Indeed, from audio equipment, i am used to the mute button being independent from the volume settings. To offer a quick way to cut the audio to zero volume and restore it back to the former volume. While the user selected volume level is not altered, i.e. restored after unmute.
The mute status shall be indicated by the icon showing the crossed out volume, whereas the Volume Label shows the user set level at any time. So the user knows to which level the volume will be restored when un-muting.
To make this behaviour consistent over the OS, the preMuteVolume is introduced to store the volume level before the mute and act as mute indicator when > zero.
For that to work, the preMuteVolume needs to be set back to zero on un-mute. afaiu.
Oh, but what about hitting the mute button while the user set zero level, hmmmm

Reply:
Ah, that's a different behavior indeed. I'll mark this as draft and think about it some more.

Reply:
OK, I have tested and I believe everything works as intended.  It might look a little more complex than necessary, but it's specifically built to handle the use case in which the user mutes, closes settings, and then re-opens the Sound page.  It should still be muted but show the correct number for the unmuted state.